### ISSUE TITLE
inappbilling plugin is not supported for .NET MAUI .NET 8.0

### ISSUE THREAD

**Issue by Ellaliu4** [jamesmontemagno/InAppBillingPlugin#645]
If you are creating an issue for a BUG please fill out this information. If you are asking a question or requesting a feature you can delete the sections below. 

**Failure to fill out this information will result in this issue being closed.** If you post a full stack trace in a bug it will be closed, please post it to http://gist.github.com and then post the link here.

## Bug1 Information

Version Number of Plugin: 8.0.5 
Device Tested On: iOS iphone 13 Pro 17.6.1
Simulator Tested On: iphone 15 ios 17.5
Version of Rider: JetBrains Rider 2024.2.7
Version of Xamarin: .Net MAUI .NET 8.0
Versions of other things you are using: 

### Steps to reproduce the Behavior
1. Install 8.0.5 inappbilling plugin, the line below greyed out.
![image](https://github.com/user-attachments/assets/3c55990e-4d41-4059-9168-c42deb998fd1)

### Expected Behavior
IsSupported should return true.
### Actual Behavior
IsSupported should return false.
### Code snippet
![image](https://github.com/user-attachments/assets/11b418ec-2a64-42d4-a650-a526d3a90d01)

## Bug2 Information

Version Number of Plugin: 9.0.2 
Device Tested On: iOS iphone 13 Pro 17.6.1
Simulator Tested On: iphone 15 ios 17.5
Version of Rider: JetBrains Rider 2024.2.7
Version of Xamarin: .Net MAUI .NET 8.0
Versions of other things you are using: 

### Steps to reproduce the Behavior
1. Install 9.0.2 inappbilling plugin, the line below greyed out.
![image](https://github.com/user-attachments/assets/81753d0f-f8ae-4936-ac59-9b914cd1853a)

### Expected Behavior
IsSupported should return true.
### Actual Behavior
IsSupported should return false.
### Code snippet
![image](https://github.com/user-attachments/assets/d389b9f5-6fdb-4af9-afe9-577b0458eeda)

**Comment by rezamohamed** [comment]
I am using the plugin on .NET Maui 8.0 just fine using 8.0.5. I am currently using it only for Android/Google play console and Apple/Apple App store. 

That being said, I am getting an error with Google Play Console when actually purchasing as of a few days (posted an issue for that here). But for your issue stated here, it does work on 8.0.

**Comment by Ellaliu4** [comment]
> I am using the plugin on .NET Maui 8.0 just fine using 8.0.5. I am currently using it only for Android/Google play console and Apple/Apple App store.

Thank you so much for replying.

**Could you kindly list the steps of how to integrate the plugin if possible or help to check below steps anything missing?**

1. add the 8.0.5 version of the InAppBilling plugin from nuget in rider in a new empty project.
2. Added the subscriptions in App store connect.
3. Enable in app purchase capibility in Xcode.
4. Ensure all the provision profile, bundle IDs are matched.
6. Add Sample code from doc in the above code sniipt to check IsSupported().
5. Test it in real iphone device with iOS 18.0.

Result: the isSupport return false.

**Comment by rezamohamed** [comment]
Yes, sure. I am not sure what the IsSupported method you have in your code is or the purpose for it. 

For starters, in the Google Play Console - Monetize - Products, create an In-App or Subscription.

This is my Service Interface to give you an idea:

```
public interface IInAppBillingService
{
    Task<InAppBillingPurchase> PurchaseAsync(string productId, ItemType itemType);
    Task<IEnumerable<InAppBillingPurchase>> GetPurchasesAsync();
    Task<IEnumerable<InAppBillingProduct>> GetProductsAsync(string[] productIds);
    Task FinalizeAndroidPurchaseAsync(InAppBillingPurchase purchase);
}
```

Let's start with just the GetProductsAsync:

```
    public async Task<IEnumerable<InAppBillingProduct>> GetProductsAsync(string[] productIds)
    {
await billing.ConnectAsync();

            if (billing == null)
                throw new Exception("Not connected to the billing service");

            // Get products for subscriptions and handle null values
            var subscriptionProducts = await billing.GetProductInfoAsync(ItemType.Subscription, productIds)
                                       ?? Enumerable.Empty<InAppBillingProduct>();

            // Get products for in-app purchases and handle null values
            var inAppPurchaseProducts = await billing.GetProductInfoAsync(ItemType.InAppPurchase, productIds)
                                        ?? Enumerable.Empty<InAppBillingProduct>();

            // Combine the results from both subscriptions and in-app purchases
             allProducts = subscriptionProducts.Concat(inAppPurchaseProducts);

await billing.DisconnectAsync();

 return allProducts;
}
```

If you are debugging Android, just make sure you created the Keystore and added it in your CSPROJ/Debug.


    <AndroidKeyStore>true</AndroidKeyStore>
    <AndroidSigningKeyStore>your.keystore</AndroidSigningKeyStore>
    <AndroidSigningKeyAlias>alias</AndroidSigningKeyAlias>
    <AndroidSigningKeyPass>pass</AndroidSigningKeyPass>
    <AndroidSigningStorePass>pass</AndroidSigningStorePass>

Make sure you have License Testing turned on in Google Play Console (Google Play - Settings - License Testing) and add the account that you will sign into the PlayStore with on your physical device (you cannot debug using a simulator).

Everything you need is in the documentation for this plugin, https://jamesmontemagno.github.io/InAppBillingPlugin/

Hope this helps!
